src/pybind/mgr/volumes/fs/operations/versions/subvolume_base.py

```diff
@@ -131,6 +131,12 @@ def _set_attrs(self, path, size, isolate_namespace, pool, uid, gid):
             # pool layout changes.
             xattr_key = 'ceph.dir.layout.pool'
             xattr_val = get_ancestor_xattr(self.fs, path, "ceph.dir.layout.pool")
+            try:
```

Comment:
Why do you need this? get_ancestor_xattr already fetches the pool_layout of the existing path. You're saving an setxattr call, but you're adding a getxattr call.

Maybe you can check if the path's layout.pool vxattr is set first, and if not set, then call get_ancestor_xattr() method. And modify get_ancestor_xattr to actually start looking from the ancestor of the path.

Reply:
> Maybe you can check if the path's layout.pool vxattr is set first, and if not set, then call get_ancestor_xattr() method. And modify get_ancestor_xattr to actually start looking from the ancestor of the path.

I had written like this initially, but code readability wasn't great. So I chose the existing way. But I didn't notice that the get_ancestor_xattr is fetching layout on the given path as oppose to the function name. I will see what can be done to fix this.

Reply:
In this particular case, where the check is required, I am sending parent path if xattr is not found on the path. So that should work and avoid extra getxattr call.